src/nomad_simulations/schema_packages/properties/band_gap.py

```diff
@@ -57,12 +57,6 @@ class ElectronicBandGap(PhysicalProperty):
         """,
     )
 
-    def __init__(
-        self, m_def: 'Section' = None, m_context: 'Context' = None, **kwargs
-    ) -> None:
-        super().__init__(m_def, m_context, **kwargs)
-        self.name = self.m_def.name
```

Comment:
Can you explain your reasoning for removing these?

Reply:
Good point. `self` ensures that it also sets the children's names correctly.
Added a test to `test_physical_properties.py` to double-check.